src/iter/parallel.rs

```diff
@@ -0,0 +1,135 @@
+use rayon::iter::ParallelIterator;
+use rayon::iter::internal::{UnindexedProducer, UnindexedConsumer, Folder, bridge_unindexed};
+use super::*;
+
+/// An `ParallelIterator` over a [`BitSetLike`] structure.
+///
+/// [`BitSetLike`]: ../../trait.BitSetLike.html
+pub struct BitParIter<T>(T);
+
+#[cfg(target_pointer_width="64")]
+pub const BITS_IN_USIZE: usize = 64;
+#[cfg(target_pointer_width="32")]
+pub const BITS_IN_USIZE: usize = 32;
+
+impl<T> BitParIter<T> {
+    /// Creates a new `BitParIter`. You usually don't call this function
+    /// but just [`.par_iter()`] on a bit set.
+    ///
+    /// [`.par_iter()`]: ../../trait.BitSetLike.html#method.par_iter
+    pub fn new(set: T) -> Self {
+        BitParIter(set)
+    }
+}
+
+impl<T> ParallelIterator for BitParIter<T>
+    where T: BitSetLike + Send + Sync,
+{
+    type Item = Index;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(BitProducer((&self.0).iter()), consumer)
+    }
+}
+
+
+/// Allows splitting and internally iterating through `BitSet`.
+///
+/// Usually used internally by `BitParIter`.
+pub struct BitProducer<'a, T: 'a + Send + Sync>(pub BitIter<&'a T>);
+
+impl<'a, T: 'a + Send + Sync> UnindexedProducer for BitProducer<'a, T>
```

Comment:
Being naive, I thought you can use rayon's stuff on any iterator. It's great to see you implementing this! We need to carefully inspect the math here, especially in the context of CI failures.